File: jodiairplus11_ui/src/services/FlightCrewAssignmentService.js
import http from "../http-common";

class FlightCrewAssignmentService {
  getAllFlightCrewAssignments(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/flightCrewAssignment/flightCrewAssignments`, searchDTO);
  }

  get(flightCrewAssignmentId) {
    return this.getRequest(`/flightCrewAssignment/${flightCrewAssignmentId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/flightCrewAssignment?field=${matchData}`, null);
  }

  addFlightCrewAssignment(data) {
    return http.post("/flightCrewAssignment/addFlightCrewAssignment", data);
  }

  update(data) {
  	return http.post("/flightCrewAssignment/updateFlightCrewAssignment", data);
  }
  
  uploadImage(data,flightCrewAssignmentId) {
  	return http.postForm("/flightCrewAssignment/uploadImage/"+flightCrewAssignmentId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new FlightCrewAssignmentService();

File: src/main/java/com/jodiairplus11/service/FlightCrewAssignmentService.java
package com.jodiairplus11.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.jodiairplus11.domain.FlightCrewAssignment;
import com.jodiairplus11.dto.FlightCrewAssignmentDTO;
import com.jodiairplus11.dto.FlightCrewAssignmentSearchDTO;
import com.jodiairplus11.dto.FlightCrewAssignmentPageDTO;
import com.jodiairplus11.dto.FlightCrewAssignmentConvertCriteriaDTO;
import com.jodiairplus11.service.GenericService;
import com.jodiairplus11.dto.common.RequestDTO;
import com.jodiairplus11.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface FlightCrewAssignmentService extends GenericService<FlightCrewAssignment, Integer> {

	List<FlightCrewAssignment> findAll();

	ResultDTO addFlightCrewAssignment(FlightCrewAssignmentDTO flightCrewAssignmentDTO, RequestDTO requestDTO);

	ResultDTO updateFlightCrewAssignment(FlightCrewAssignmentDTO flightCrewAssignmentDTO, RequestDTO requestDTO);

    Page<FlightCrewAssignment> getAllFlightCrewAssignments(Pageable pageable);

    Page<FlightCrewAssignment> getAllFlightCrewAssignments(Specification<FlightCrewAssignment> spec, Pageable pageable);

	ResponseEntity<FlightCrewAssignmentPageDTO> getFlightCrewAssignments(FlightCrewAssignmentSearchDTO flightCrewAssignmentSearchDTO);
	
	List<FlightCrewAssignmentDTO> convertFlightCrewAssignmentsToFlightCrewAssignmentDTOs(List<FlightCrewAssignment> flightCrewAssignments, FlightCrewAssignmentConvertCriteriaDTO convertCriteria);

	FlightCrewAssignmentDTO getFlightCrewAssignmentDTOById(Integer flightCrewAssignmentId);







}





